akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -518,19 +606,94 @@ final class ClusterReceptionistSettings(
   def withResponseTunnelReceiveTimeout(responseTunnelReceiveTimeout: FiniteDuration): ClusterReceptionistSettings =
     copy(responseTunnelReceiveTimeout = responseTunnelReceiveTimeout)
 
+  def withHeartbeat(
+    heartbeatInterval: FiniteDuration,
+    acceptableHeartbeatPause: FiniteDuration,
+    failureDetectionInterval: FiniteDuration): ClusterReceptionistSettings =
+    copy(
+      heartbeatInterval = heartbeatInterval,
+      acceptableHeartbeatPause = acceptableHeartbeatPause,
+      failureDetectionInterval = failureDetectionInterval)
+
   private def copy(
     role: Option[String] = role,
     numberOfContacts: Int = numberOfContacts,
-    responseTunnelReceiveTimeout: FiniteDuration = responseTunnelReceiveTimeout): ClusterReceptionistSettings =
-    new ClusterReceptionistSettings(role, numberOfContacts, responseTunnelReceiveTimeout)
-
+    responseTunnelReceiveTimeout: FiniteDuration = responseTunnelReceiveTimeout,
+    heartbeatInterval: FiniteDuration = heartbeatInterval,
+    acceptableHeartbeatPause: FiniteDuration = acceptableHeartbeatPause,
+    failureDetectionInterval: FiniteDuration = failureDetectionInterval): ClusterReceptionistSettings =
+    new ClusterReceptionistSettings(
+      role,
+      numberOfContacts,
+      responseTunnelReceiveTimeout,
+      heartbeatInterval,
+      acceptableHeartbeatPause,
+      failureDetectionInterval)
 }
 
 /**
  * Marker trait for remote messages with special serializer.
  */
 sealed trait ClusterClientMessage extends Serializable
 
+/**
+ * Declares a super type for all events emitted by the `ClusterReceptionist`.
+ * in relation to cluster clients being interacted with.
+ * Events are published to the Akka event stream (system.eventStream). Given
+ * that these messages are transmitted across the event bus there is no need
+ * for serialization.
+ */
+sealed trait ClusterClientInteraction extends NoSerializationVerificationNeeded {
+  val clusterClient: ActorRef
+  val clusterReceptionist: ActorRef
+}
```

Comment:
It would be super to use some marker trait to ensure that the message is destined for remoting, or not. Here, I'd like to enforce that the message is not suitable for remoting purposes given that it doesn't need to be.